etcd/src/main/scala/com/github/fit51/reactiveconfig/etcd/EtcdConfigStorage.scala

```diff
@@ -34,25 +38,30 @@ class EtcdConfigStorage[F[_]: Async: ContextShift, ParsedData](etcd: EtcdClient[
   private val storage: TrieMap[String, Value[ParsedData]] = TrieMap.empty
 
   override def load(): F[TrieMap[String, Value[ParsedData]]] =
-    etcd
-      .getRecursiveSinceRevision(prefix, revision)
-      .map {
-        case (kvs, rev) =>
-          revision = rev
-          logger.info(s"EtcdConfig: Updated to rev: $rev")
-          kvs.foreach(saveKeyValue(_, checkVersions = true))
-          storage
-      }
+    prefixes.traverse { prefix =>
+      etcd
+        .getRecursiveSinceRevision(prefix, revision)
+        .map {
+          case (kvs, rev) =>
+            revision = rev
```

Comment:
Shared variable across several different prefixes. You probably should use `Map[String, Long]` instead of one `Long`.

Reply:
revision is not handled correctly, even if there is only one prefix. There is no guarantee, kv-s would come in asc order of revision.
We do not actually use revision anyway, I would remove line 46 for now and return to it in #11